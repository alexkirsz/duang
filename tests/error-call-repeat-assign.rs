// this test check the error message when user repeatedly assign to same parameter.
use duang::duang;

duang!(
pub fn foo<T>(a: T,b: f64 = 13.0, c: T = a*a) -> (T,f64,T)
where
  T: std::ops::Mul<T, Output = T>,
  T: std::fmt::Display,
  T: Copy,
{
  (a,b,c)
}
);

fn main() {
  foo!(1,c=1,c=1);
  foo!(1,c=1,a=1);
}
