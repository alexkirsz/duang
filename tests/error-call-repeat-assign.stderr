error: This argument is assigned more than once.
  --> $DIR/error-call-repeat-assign.rs:16:14
   |
16 |   foo!(1,c=1,c=1);
   |   -----------^---- in this macro invocation

error: Previously assigned here
  --> $DIR/error-call-repeat-assign.rs:16:10
   |
16 |   foo!(1,c=1,c=1);
   |   -------^-------- in this macro invocation

error: This argument is assigned more than once.
  --> $DIR/error-call-repeat-assign.rs:17:14
   |
17 |   foo!(1,c=1,a=1);
   |   -----------^---- in this macro invocation

error: Previously assigned here
  --> $DIR/error-call-repeat-assign.rs:17:8
   |
17 |   foo!(1,c=1,a=1);
   |   -----^---------- in this macro invocation
